src/ot_orchestration/dags/config/platform.py

```diff
@@ -0,0 +1,131 @@
+
+from pathlib import Path
+from typing import Any
+
+from ot_orchestration.utils import read_hocon_config, read_yaml_config
+
+
+class PlatformConfig:
+    """Configuration class for the platform pipeline.
+
+    This class reads the configuration files for the platform dag, PIS and ETL
+    applications, performs some operations on them and then exposes the values.
+
+    Some fields in PIS and ETL application configuration files are replaced with
+    values from the platform dag configuration, which is the only one the user of
+    the orchestrator has to modify to run the pipeline.
+
+    The configuration files are expected to be in the same directory as this file.
+    They are:
+    - `platform.yaml`: contains the general configuration for the pipeline.
+    - `pis.yaml`: contains the configuration for the PIS steps.
+    - `etl.conf`: contains the configuration for the ETL steps.
+    """
+
+    def __init__(self) -> None:
+        self.platform_config_path = Path(__file__).parent / "platform.yaml"
+        self.pis_config_local_path = Path(__file__).parent / "pis.yaml"
+        self.etl_config_local_path = Path(__file__).parent / "etl.conf"
+
+        # These are hardcoded config values that are not meant to change often.
+        # It is safe to extract them from here into the config file in case they
+        # become more dynamic. Just add them to self in the initialization so they
+        # are available.
+
+        # fmt: off
+        # The base image for PIS, the version tag will be appended from the config file.
+        pis_image_base = "europe-west1-docker.pkg.dev/open-targets-eu-dev/platform-input-support-test/platform-input-support-test"
+        # The base url for the ETL jar, the version will be replaced in from the config file.
+        etl_jar_base = "https://github.com/opentargets/platform-etl-backend/releases/download/v{version}/etl-backend-{version}.jar"
+
+        # The disk size for PIS vms, in GB.
+        # Note: although not all steps need this much space, it is easier to have a
+        # single value for all steps, and the machines are so short-lived that it
+        # doesn't matter much with respect to cost.
+        self.pis_disk_size = 150
+
+        # The service account and scopes to use (only used by PIS so far).
+        # The drive scope is needed to download spreadsheets from Google Drive
+        # for the PIS otar step.
+        self.service_account = "platform-input-support@open-targets-eu-dev.iam.gserviceaccount.com"
+        self.service_account_scopes = ["https://www.googleapis.com/auth/drive"]
+        # fmt: on
+
+        # Platform pipeline settings.
+        settings = read_yaml_config(self.platform_config_path)
+        self.gcs_url = settings["gcs_url"]
+        self.chembl_version = settings["chembl_version"]
+        self.efo_version = settings["efo_version"]
+        self.ensembl_version = settings["ensembl_version"]
+        self.is_ppp = settings["is_ppp"]
```

Comment:
I was not worrying very much about PPP. The mechanism to run its pipeline needs some reevaluation altogether.

In the current process we use at the moment for releases,  there are some different steps: we copy files from the platform run of PIS instead of running it again so they are exactly the same.

Maybe we can have two copies of the dag altogether? As I said, this is just an initial approach but we have to think it through.

Reply:
Lets think about it later.